worldedit-cli/src/main/java/com/sk89q/worldedit/cli/CLIWorldEdit.java

```diff
@@ -0,0 +1,349 @@
+        }
+        // Tags
+        for (String name : fileRegistries.getDataFile().blocktags.keySet()) {
+            if (BlockCategory.REGISTRY.get(name) == null) {
+                BlockCategory.REGISTRY.register(name, new BlockCategory(name));
+            }
+        }
+        for (String name : fileRegistries.getDataFile().itemtags.keySet()) {
+            if (ItemCategory.REGISTRY.get(name) == null) {
+                ItemCategory.REGISTRY.register(name, new ItemCategory(name));
+            }
+        }
+    }
+
+    public void onInitialized() {
+        // Setup working directory
+        workingDir = new File("worldedit").toPath();
+        if (!Files.exists(workingDir)) {
+            try {
+                Files.createDirectory(workingDir);
+            } catch (IOException e) {
+                throw new UncheckedIOException(e);
+            }
+        }
+
+        this.platform = new CLIPlatform(this);
+        LOGGER.info("WorldEdit CLI (version " + getInternalVersion() + ") is loaded");
+    }
+
+    public void onStarted() {
+        setupPlatform();
+
+        setupRegistries();
+        WorldEdit.getInstance().loadMappings();
+
+        config = new CLIConfiguration(this);
+        config.load();
+        commandSender = new CLICommandSender(this, LOGGER);
+
+        WorldEdit.getInstance().getEventBus().post(new PlatformReadyEvent());
+    }
+
+    public void onStopped() {
+        WorldEdit worldEdit = WorldEdit.getInstance();
+        worldEdit.getSessionManager().unload();
+        worldEdit.getPlatformManager().unregister(platform);
+    }
+
+    public FileRegistries getFileRegistries() {
+        return this.fileRegistries;
+    }
+
+    /**
+     * Get the configuration.
+     *
+     * @return the CLI configuration
+     */
+    CLIConfiguration getConfig() {
+        return this.config;
+    }
+
+    /**
+     * Get the WorldEdit proxy for the platform.
+     *
+     * @return the WorldEdit platform
+     */
+    public Platform getPlatform() {
+        return this.platform;
+    }
+
+    /**
+     * Get the working directory where WorldEdit's files are stored.
+     *
+     * @return the working directory
+     */
+    public File getWorkingDir() {
+        return this.workingDir.toFile();
+    }
+
+    /**
+     * Get the version of the WorldEdit-CLI implementation.
+     *
+     * @return a version string
+     */
+    String getInternalVersion() {
+        if (version == null) {
+            version = getClass().getPackage().getImplementationVersion();
+        }
+        return version;
+    }
+
+    public void run(InputStream inputStream) {
+        Scanner scanner = new Scanner(inputStream);
+        while (scanner.hasNextLine()) {
+            String line = scanner.nextLine();
+            if (line.equalsIgnoreCase("stop")) {
+                commandSender.print("Stopping!");
+                break;
+            }
+            if (line.startsWith("save")) {
+                String[] bits = line.split(" ");
```

Comment:
I would prefer it if we allowed quoting or escaping here, just for user-friendliness.